presto-main/src/main/java/com/facebook/presto/cost/PlanNodeStatsEstimateMath.java

```diff
@@ -14,86 +14,130 @@
 package com.facebook.presto.cost;
 
 import static com.google.common.base.Preconditions.checkArgument;
+import static java.lang.Double.NaN;
+import static java.lang.Double.isNaN;
+import static java.lang.Double.max;
+import static java.lang.Double.min;
 import static java.util.stream.Stream.concat;
 
 public class PlanNodeStatsEstimateMath
 {
     private PlanNodeStatsEstimateMath()
     {}
 
-    @FunctionalInterface
-    private interface RangeSubtractionStrategy
+    public static PlanNodeStatsEstimate subtractStats(PlanNodeStatsEstimate superset, PlanNodeStatsEstimate subset)
```

Comment:
Please make name more expressive, e.g: `subtractSubsetStats` and add Javadoc.

Reply:
Any ideas how javadoc should look like without replicating the code logic?

Reply:
Maybe:
```
    /**
     * Subtracts subset stats from supersets stats. It is assumed that each NDV from subset
     * has a matching NDV in superset.
     */
```
but it's a nit